src/test/java/cloud/eppo/EppoClientTest.java

```diff
@@ -0,0 +1,449 @@
+package cloud.eppo;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThrows;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import cloud.eppo.helpers.AssignmentTestCase;
+import cloud.eppo.helpers.AssignmentTestCaseDeserializer;
+import cloud.eppo.helpers.SubjectAssignment;
+import cloud.eppo.helpers.TestCaseValue;
+import cloud.eppo.logging.Assignment;
+import cloud.eppo.logging.AssignmentLogger;
+import cloud.eppo.ufc.dto.EppoValue;
+import cloud.eppo.ufc.dto.SubjectAttributes;
+import cloud.eppo.ufc.dto.VariationType;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.*;
+import java.util.stream.Stream;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.ArgumentCaptor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class EppoClientTest {
```

Comment:
Adapted from Android's implementation ([source](https://github.com/Eppo-exp/android-sdk/blob/main/eppo/src/androidTest/java/cloud/eppo/android/EppoClientTest.java)) with a few modifications:
* Everything is synchronous (for now)
* Anything related to caching has been removed (for now)
* `@ParameterizedTest` just uses the file (as opposed to the whole test case) for a nice display of the test names in the IDE
* JSON-related tests use Jackson instead of built-in/Android JSON